app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt

```diff
@@ -192,6 +193,21 @@ class GleanMetricsService(private val context: Context) : MetricsService {
     }
 
     override fun shouldTrack(event: Event): Boolean {
+        if (event is Event.PreferenceToggled) {
+            return switchPreferenceTelemetryAllowList.contains(event.preferenceKey)
+        }
+
         return event.wrapper != null
     }
+
+    // Switch preferences that should have their toggle events sent to Glean
+    private val switchPreferenceTelemetryAllowList = listOf(
+        context.getString(R.string.pref_key_leakcanary),
+        context.getString(R.string.pref_key_make_default_browser),
+        context.getString(R.string.pref_key_show_search_suggestions),
+        context.getString(R.string.pref_key_show_visited_sites_bookmarks),
+        context.getString(R.string.pref_key_remote_debugging),
+        context.getString(R.string.pref_key_telemetry),
```

Comment:
I guess technically we won't ever receive this telemetry getting toggled off, but not important enough to exclude.